### PR TITLE
Release: decK 1.28

### PR DESCRIPTION
**Pull request by lena-larionova**
### Description

Release 1.28.x decK docs MVP. CLI reference updates only.
Guides to follow.

### Testing instructions

Preview link: https://deploy-preview-6420--kongdocs.netlify.app/deck/latest/reference/deck/

### Checklist 

- [x] Review label added 
- [x] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

### CODE REVIEW

File: app/_src/deck/reference/deck.md [hunk 1]
```diff
@@ -13,7 +13,54 @@ It can be used to export, import, or sync entities to Kong.
...
+* [deck file convert](/deck/{{page.kong_version}}/reference/deck_file_convert)	 - Convert files from one format into another format
+* [deck file lint](/deck/{{page.kong_version}}/reference/deck_file_lint)	 - Lint a file against a ruleset
+* [deck file list-tags](/deck/{{page.kong_version}}/reference/deck_file_list-tags)	 - List current tags from objects in a decK file
+* [deck file merge](/deck/{{page.kong_version}}/reference/deck_file_merge)	 - Merge multiple decK files into one
+* [deck file openapi2kong](/deck/{{page.kong_version}}/reference/deck_file_openapi2kong)	 - Convert OpenAPI files to Kong's decK format
+* [deck file patch](/deck/{{page.kong_version}}/reference/deck_file_patch)	 - Apply patches on top of a decK file
+* [deck file remove-tags](/deck/{{page.kong_version}}/reference/deck_file_remove-tags)	 - Remove tags from objects in a decK file
+* [deck file render](/deck/{{page.kong_version}}/reference/deck_file_render)	 - Combines multiple complete configuration files and renders them as one Kong declarative config file.
+* [deck file validate](/deck/{{page.kong_version}}/reference/deck_file_validate)	 - Validate the state file (locally)
```
Comment (rspurgeon) [inline app/_src/deck/reference/deck.md:47]:
```suggestion
* [deck file validate](/deck/{{page.kong_version}}/reference/deck_file_validate)	 - Locally validates the state file for basic structure or relationship errors.
```

File: app/_src/deck/reference/deck.md [hunk 2]
```diff
@@ -13,7 +13,54 @@ It can be used to export, import, or sync entities to Kong.
...
+General:
+* [deck completion](/deck/{{page.kong_version}}/reference/deck_completion/)	 - Generate completion script
+* [deck version](/deck/{{page.kong_version}}/reference/deck_version/) - Print the decK version
+
+
+Gateway subcommands:
+* [deck gateway diff](/deck/{{page.kong_version}}/reference/deck_gateway_diff/)	- Diff the current entities in Kong with the one on disks
+* [deck gateway dump](/deck/{{page.kong_version}}/reference/deck_gateway_dump/)	- Export Kong configuration to a file
+* [deck gateway ping](/deck/{{page.kong_version}}/reference/deck_gateway_ping/)	- Verify connectivity with Kong
+* [deck gateway reset](/deck/{{page.kong_version}}/reference/deck_gateway_reset/) - Reset deletes all entities in Kong
```
Comment (rspurgeon) [inline app/_src/deck/reference/deck.md:31]:
```suggestion
* [deck gateway reset](/deck/{{page.kong_version}}/reference/deck_gateway_reset/) - Deletes all entities in Kong
```

File: app/_src/deck/reference/deck.md [hunk 3]
```diff
@@ -13,7 +13,54 @@ It can be used to export, import, or sync entities to Kong.
...
+* [deck gateway sync](/deck/{{page.kong_version}}/reference/deck_gateway_sync/)	- Sync performs operations to get Kong's configuration to match the state file
+* [deck gateway validate](/deck/{{page.kong_version}}/reference/deck_gateway_validate/)	- Validate the state file
+
+
+File subcommands:
+* [deck file add-plugins](/deck/{{page.kong_version}}/reference/deck_file_add-plugins)	 - Add plugins to objects in a decK file
+* [deck file add-tags](/deck/{{page.kong_version}}/reference/deck_file_add-tags)	 - Add tags to objects in a decK file
+* [deck file convert](/deck/{{page.kong_version}}/reference/deck_file_convert)	 - Convert files from one format into another format
+* [deck file lint](/deck/{{page.kong_version}}/reference/deck_file_lint)	 - Lint a file against a ruleset
```
Comment (rspurgeon) [inline app/_src/deck/reference/deck.md:40]:
```suggestion
* [deck file lint](/deck/{{page.kong_version}}/reference/deck_file_lint)	 - Validate a file against a ruleset
```

File: app/_src/deck/reference/deck.md [hunk 4]
```diff
@@ -13,7 +13,54 @@ It can be used to export, import, or sync entities to Kong.
...
+* [deck file list-tags](/deck/{{page.kong_version}}/reference/deck_file_list-tags)	 - List current tags from objects in a decK file
+* [deck file merge](/deck/{{page.kong_version}}/reference/deck_file_merge)	 - Merge multiple decK files into one
+* [deck file openapi2kong](/deck/{{page.kong_version}}/reference/deck_file_openapi2kong)	 - Convert OpenAPI files to Kong's decK format
```
Comment (rspurgeon) [inline app/_src/deck/reference/deck.md:43]:
```suggestion
* [deck file openapi2kong](/deck/{{page.kong_version}}/reference/deck_file_openapi2kong)	 - Convert OpenAPI specifications to Kong's decK format
```

File: app/_src/deck/reference/deck_file_lint.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
+---
+title: deck file lint
+source_url: https://github.com/Kong/deck/tree/main/cmd/file_lint.go
+content_type: reference
+---
+
+Validate a decK state file against a linting ruleset, reporting any violations or failures.
+Report output can be returned in JSON, YAML, or human readable format (see --format).
+Ruleset Docs: https://quobix.com/vacuum/rulesets/
```
Comment (rspurgeon) [inline app/_src/deck/reference/deck_file_lint.md:9]: can this be clickable?

File: app/_src/deck/reference/deck_diff.md [hunk 6]
```diff
@@ -36,7 +42,7 @@ and exit code 1 if an error occurs. (Default: `false`)
 :  Maximum number of concurrent operations. (Default: `10`)
 
 `--rbac-resources-only`
-:  sync only the RBAC resources (Kong Enterprise only). (Default: `false`)
+:  sync only the RBAC resources ({{site.ee_product_name}} only). (Default: `false`)
```
Comment (Guaris) [inline app/_src/deck/reference/deck_diff.md:45]:
```suggestion
:  sync only the RBAC resources **{{site.ee_product_name}} only**. (Default: `false`)
```
Not really anything to do with your PR but the parenthesis back to back looks weird to me.

File: app/_src/deck/reference/deck_diff.md [hunk 7]
```diff
@@ -65,7 +71,7 @@ This flag can be specified multiple times for multiple files.
 Use `-` to read from stdin. (Default: `[kong.yaml]`)
 
 `-w`, `--workspace`
-:  Diff configuration with a specific workspace (Kong Enterprise only).
+:  Diff configuration with a specific workspace ({{site.ee_product_name}} only).
 This takes precedence over _workspace fields in state files.
```
Comment (Guaris) [inline app/_src/deck/reference/deck_diff.md:75]: Is `_workspace` a type of field? Or is that a stray underscore?

File: app/_src/deck/reference/deck_dump.md [hunk 8]
```diff
@@ -19,19 +24,19 @@ deck dump [command-specific flags] [global flags]
 ## Flags
 
 `--all-workspaces`
-:  dump configuration of all Workspaces (Kong Enterprise only). (Default: `false`)
+:  dump configuration of all Workspaces ({{site.ee_product_name}} only). (Default: `false`)
```
Comment (Guaris) [inline app/_src/deck/reference/deck_dump.md:27]:
```suggestion
:  dump configuration of all workspaces **{{site.ee_product_name}} only**. (Default: `false`)
```

File: app/_src/deck/reference/deck_dump.md [hunk 9]
```diff
@@ -19,19 +24,19 @@ deck dump [command-specific flags] [global flags]
...
 `-h`, `--help`
-:  help for dump (Default: `false`)
+:  help for dump 
 
 `-o`, `--output-file`
 :  file to which to write Kong's configuration.Use `-` to write to stdout. (Default: `"kong"`)
 
 `--rbac-resources-only`
-:  export only the RBAC resources (Kong Enterprise only). (Default: `false`)
+:  export only the RBAC resources ({{site.ee_product_name}} only). (Default: `false`)
```
Comment (Guaris) [inline app/_src/deck/reference/deck_dump.md:39]:
```suggestion
:  export only the RBAC resources **{{site.ee_product_name}} only**. (Default: `false`)
```

File: app/_src/deck/reference/deck_dump.md [hunk 10]
```diff
@@ -62,7 +67,7 @@ When this setting has multiple tag values, entities must match every tag.
 :  write ID of all entities in the output (Default: `false`)
 
 `-w`, `--workspace`
-:  dump configuration of a specific Workspace(Kong Enterprise only).
+:  dump configuration of a specific Workspace({{site.ee_product_name}} only).
```
Comment (Guaris) [inline app/_src/deck/reference/deck_dump.md:70]:
```suggestion
:  dump configuration of a specific workspace **{{site.ee_product_name}} only**.
```

File: app/_src/deck/reference/deck_file.md [hunk 11]
```diff
@@ -15,14 +15,20 @@ Subcommand to host the decK file manipulation operations.
...
 
-## See also
+## decK file commands
 
-* [deck](/deck/{{page.kong_version}}/reference/deck)	 - Administer your Kong clusters declaratively
 * [deck file add-plugins](/deck/{{page.kong_version}}/reference/deck_file_add-plugins)	 - Add plugins to objects in a decK file
 * [deck file add-tags](/deck/{{page.kong_version}}/reference/deck_file_add-tags)	 - Add tags to objects in a decK file
+{% if_version gte:1.28.x %}
+* [deck file convert](/deck/{{page.kong_version}}/reference/deck_file_convert)	 - Convert files from one format into another format
+* [deck file lint](/deck/{{page.kong_version}}/reference/deck_file_lint)	 - Lint a file against a ruleset
```
Comment (Guaris) [inline app/_src/deck/reference/deck_file.md:24]:
```suggestion
* [deck file lint](/deck/{{page.kong_version}}/reference/deck_file_lint)	 - Validate a file against a ruleset
```

File: app/_src/deck/reference/deck_file.md [hunk 12]
```diff
@@ -15,14 +15,20 @@ Subcommand to host the decK file manipulation operations.
...
+{% endif_version %}
 * [deck file list-tags](/deck/{{page.kong_version}}/reference/deck_file_list-tags)	 - List current tags from objects in a decK file
 * [deck file merge](/deck/{{page.kong_version}}/reference/deck_file_merge)	 - Merge multiple decK files into one
 * [deck file openapi2kong](/deck/{{page.kong_version}}/reference/deck_file_openapi2kong)	 - Convert OpenAPI files to Kong's decK format
 * [deck file patch](/deck/{{page.kong_version}}/reference/deck_file_patch)	 - Apply patches on top of a decK file
 * [deck file remove-tags](/deck/{{page.kong_version}}/reference/deck_file_remove-tags)	 - Remove tags from objects in a decK file
-
+{% if_version gte:1.28.x %}
+* [deck file render](/deck/{{page.kong_version}}/reference/deck_file_render)	 - Combines multiple complete configuration files and renders them as one Kong declarative config file.
+* [deck file validate](/deck/{{page.kong_version}}/reference/deck_file_validate)	 - Validate the state file (locally)
```
Comment (Guaris) [inline app/_src/deck/reference/deck_file.md:33]:
```suggestion
* [deck file validate](/deck/{{page.kong_version}}/reference/deck_file_validate)	 - Validate the state file locally
```

File: app/_src/deck/reference/deck_file_add-plugins.md [hunk 13]
```diff
@@ -35,6 +35,16 @@ order they are given:
 deck file add-plugins [command-specific flags] [global flags] [...plugin-files]
 ```
 
+## Examples
+
+```
+# adds 2 plugins to all services in a deck file, except if they are already present
```
Comment (Guaris) [inline app/_src/deck/reference/deck_file_add-plugins.md:41]:
```suggestion
# adds 2 plugins to all services in a deck file, unless they are already present
```

File: app/_src/deck/reference/deck_file_lint.md (hunk 5 above)
Comment (Guaris) [inline app/_src/deck/reference/deck_file_lint.md:9]:
```suggestion
[Ruleset Docs](https://quobix.com/vacuum/rulesets/)
```

File: app/_src/deck/reference/deck_file_render.md [hunk 14]
```diff
@@ -3,7 +3,23 @@ title: deck file render
...
-Render the configuration as Kong declarative config.
+Combines multiple complete configuration files and renders them as one Kong
+declarative config file.
+
+This command renders a full declarative configuration in JSON or YAML format by assembling 
+multiple files and populating defaults and environment substitutions. 
+This command is useful to observe what configuration would be sent prior to synchronizing to 
+the gateway.
+ 
+In comparison to the "deck file merge" command, the render command accepts
```
Comment (Guaris) [inline app/_src/deck/reference/deck_file_render.md:14]:
```suggestion
In comparison to the `deck file merge` command, the render command accepts 
```

File: app/_src/deck/reference/deck_file_render.md [hunk 15]
```diff
@@ -3,7 +3,23 @@ title: deck file render
...
+Combines multiple complete configuration files and renders them as one Kong
+declarative config file.
+
+This command renders a full declarative configuration in JSON or YAML format by assembling 
+multiple files and populating defaults and environment substitutions. 
+This command is useful to observe what configuration would be sent prior to synchronizing to 
+the gateway.
+ 
+In comparison to the "deck file merge" command, the render command accepts 
+complete configuration files, while "deck file merge" can operate on partial files.
```
Comment (Guaris) [inline app/_src/deck/reference/deck_file_render.md:15]:
```suggestion
complete configuration files, while `deck file merge` can operate on partial files.
```

File: app/_src/deck/reference/deck_file_validate.md [hunk 16]
```diff
@@ -0,0 +1,49 @@
+---
+title: deck file validate
+source_url: https://github.com/Kong/deck/tree/main/cmd/file_validate.go
+content_type: reference
+---
+
+The validate command reads the state file and ensures validity.
```
Comment (Guaris) [inline app/_src/deck/reference/deck_file_validate.md:7]:
```suggestion
The validate command reads the state file and ensures its validity.
```

File: app/_src/deck/reference/deck_file_validate.md [hunk 17]
```diff
@@ -0,0 +1,49 @@
...
+It reads all the specified state files and reports YAML/JSON
```
Comment (Guaris) [inline app/_src/deck/reference/deck_file_validate.md:8]:
```suggestion
It reads all the specified state files and reports any YAML/JSON
```

File: app/_src/deck/reference/deck_file_validate.md [hunk 18]
```diff
@@ -0,0 +1,49 @@
...
+parsing issues. It also checks for foreign relationships
```
Comment (Guaris) [inline app/_src/deck/reference/deck_file_validate.md:9]:
```suggestion
parsing issues. Additionally, it checks for foreign relationships
```

File: app/_src/deck/reference/deck_file_validate.md [hunk 19]
```diff
@@ -0,0 +1,49 @@
...
+and alerts if there are broken relationships, or missing links present.
```
Comment (Guaris) [inline app/_src/deck/reference/deck_file_validate.md:10]:
```suggestion
and alerts in cases of broken relationships or missing links.
```

File: app/_src/deck/reference/deck_file_validate.md [hunk 20]
```diff
@@ -0,0 +1,49 @@
...
+
+No communication takes places between decK and Kong during the execution of
+this command. This is faster than the online validation, but catches fewer errors.
```
Comment (Guaris) [inline app/_src/deck/reference/deck_file_validate.md:13]:
```suggestion
this command. This process is faster than online validation but may catch fewer errors.
```

File: app/_src/deck/reference/deck_gateway_diff.md [hunk 21]
```diff
@@ -0,0 +1,68 @@
...
+## Flags
+
+`-h`, `--help`
+:  help for diff 
+
+`--json-output`
+:  generate command execution report in a JSON format (Default: `false`)
+
+`--no-mask-deck-env-vars-value`
+:  do not mask DECK_ environment variable values at diff output. (Default: `false`)
```
Comment (Guaris) [inline app/_src/deck/reference/deck_gateway_diff.md:29]:
```suggestion
:  do not mask DECK_ environment variable values in the diff output. (Default: `false`)
```

File: app/_src/deck/reference/deck_gateway_dump.md [hunk 22]
```diff
@@ -0,0 +1,64 @@
...
+When this setting has multiple tag values, entities must match every tag.
+
+`--skip-ca-certificates`
+:  do not dump CA certificates. (Default: `false`)
+
+`--skip-consumers`
+:  skip exporting consumers, consumer-groups and any plugins associated with them. (Default: `false`)
+
+`--with-id`
+:  write ID of all entities in the output (Default: `false`)
```
Comment (Guaris) [inline app/_src/deck/reference/deck_gateway_dump.md:47]:
```suggestion
:  return the ID of all entities in the output (Default: `false`)
```

File: app/_src/deck/reference/deck_gateway_dump.md [hunk 23]
```diff
@@ -0,0 +1,64 @@
...
+
+`-w`, `--workspace`
+:  dump configuration of a specific Workspace({{site.ee_product_name}} only).
```
Comment (Guaris) [inline app/_src/deck/reference/deck_gateway_dump.md:50]:
```suggestion
:  dump configuration of a specific workspace ({{site.ee_product_name}} only).
```

File: app/_src/deck/reference/deck_gateway_reset.md [hunk 24]
```diff
@@ -0,0 +1,62 @@
...
+:  Skip interactive confirmation prompt before reset. (Default: `false`)
+
+`-h`, `--help`
+:  help for reset 
+
+`--json-output`
+:  generate command execution report in a JSON format (Default: `false`)
+
+`--no-mask-deck-env-vars-value`
+:  do not mask DECK_ environment variable values at diff output. (Default: `false`)
```
Comment (Guaris) [inline app/_src/deck/reference/deck_gateway_reset.md:35]:
```suggestion
:  do not mask DECK_ environment variable values in the diff output. (Default: `false`)
```

File: app/_src/deck/reference/deck_gateway_sync.md [hunk 25]
```diff
@@ -0,0 +1,65 @@
...
+See `db_update_propagation` in kong.conf. (Default: `0`)
+
+`-h`, `--help`
+:  help for sync 
+
+`--json-output`
+:  generate command execution report in a JSON format (Default: `false`)
+
+`--no-mask-deck-env-vars-value`
+:  do not mask DECK_ environment variable values at diff output. (Default: `false`)
```
Comment (Guaris) [inline app/_src/deck/reference/deck_gateway_sync.md:30]:
```suggestion
:  do not mask DECK_ environment variable values in the diff output. (Default: `false`)
```

File: app/_src/deck/reference/deck_konnect_diff.md [hunk 26]
```diff
@@ -24,28 +24,28 @@ deck konnect diff [command-specific flags] [global flags]
 ## Flags
 
 `-h`, `--help`
-:  help for diff (Default: `false`)
+:  help for diff 
 
 {% if_version gte:1.16.x %}
 `--no-mask-deck-env-vars-value`
-:  do not mask `DECK_` environment variable values at diff output. (Default: `false`)
+:  do not mask `DECK_` environment variable values at diff output.
```
Comment (Guaris) [inline app/_src/deck/reference/deck_konnect_diff.md:31]:
```suggestion
:  do not mask `DECK_` environment variable values in the diff output. 
```

File: app/_src/deck/reference/deck_reset.md [hunk 27]
```diff
@@ -20,21 +25,21 @@ deck reset [command-specific flags] [global flags]
...
 `-f`, `--force`
 :  Skip interactive confirmation prompt before reset. (Default: `false`)
 
 `-h`, `--help`
-:  help for reset (Default: `false`)
+:  help for reset 
 
 {% if_version gte:1.16.x %}
 `--no-mask-deck-env-vars-value`
 :  do not mask `DECK_` environment variable values at diff output. (Default: `false`)
```
Comment (Guaris) [inline app/_src/deck/reference/deck_reset.md:38]:
```suggestion
:  do not mask `DECK_` environment variable values in the diff output. (Default: `false`)
```

**Review by Guaris — CHANGES_REQUESTED** [review]
Would it make sense to make the deck gateway commands section an include? 
```
## decK gateway commands

* [deck](/deck/{{page.kong_version}}/reference/deck)	 - Administer your Kong clusters declaratively
...
```

File: app/_src/deck/reference/deck_diff.md (hunk 7 above)
Comment (rspurgeon) [reply app/_src/deck/reference/deck_diff.md:75]: `_workspace` is accurate : https://docs.konghq.com/deck/latest/guides/kong-enterprise/#manage-one-workspace-at-a-time

File: app/_src/deck/reference/deck_file.md (hunk 11 above)
Comment (lena-larionova) [reply app/_src/deck/reference/deck_file.md:24]: Leaving this one as is, since there's a separate `validate` command.

**Comment by lena-larionova** [comment]
@rspurgeon with the 1.28.1 patch out, do you think this is safe to publish? I just heard from Hayk in a Gateway sync call that tests are passing with this new patch, though you might want to double-check my hearsay.

**Comment by rspurgeon** [comment]
> @rspurgeon with the 1.28.1 patch out, do you think this is safe to publish? I just heard from Hayk in a Gateway sync call that tests are passing with this new patch, though you might want to double-check my hearsay.

Yes, I think we are good with this

**Review by Guaris — APPROVED** [review]

**Pull request merged**